packages/ui/src/Banner.tsx

```diff
@@ -16,141 +94,107 @@ export const hideBanner = (id: string): Promise<void> => {
-    }
-  };
-
   if (!show) {
     return null;
   }
 
-  if ((type === ACTION && !onClick) || (type === CUSTOM_BUTTON && !onClick)) {
-    console.warn("Banners with type action require an onClick property.");
-  }
-
-  const negativeMargin = negativeXMargin * -4;
-
   return (
-    <Box
-      color={color}
-      dangerouslySetInlineStyle={{
-        __style: {
-          marginLeft: negativeMargin,
-          marginRight: negativeMargin,
-        },
+    <View
+      style={{
+        alignItems: "center",
+        backgroundColor: theme.surface[bgColor],
+        height: "auto",
+        width: "100%",
+        margin: "auto",
+        flexDirection: "row",
+        borderRadius: theme.radius.default as any,
+        minHeight: theme.spacing.xl as any,
+        padding: theme.spacing.xs as any,
       }}
-      direction="row"
-      justifyContent="between"
-      paddingX={3}
-      paddingY={2}
-      rounding={shape}
-      shadow
-      width={width || Unifier.utils.dimensions().width || "100%"}
-      onClick={type === DISMISS ? dismiss : undefined}
     >
-      {iconName && (
-        <Box justifyContent="center" width={32}>
-          <Icon color={textColor} iconName={iconName} size="lg" />
-        </Box>
-      )}
-      <Box alignItems="center" direction="column" flex="shrink" justifyContent="center">
-        <Box paddingY={1}>
-          <Text align="center" bold color={textColor}>
-            {text}
-          </Text>
-        </Box>
-        {subtext && (
-          <Box paddingY={1}>
-            <Text align="center" color={textColor}>
-              {subtext}
-            </Text>
-          </Box>
+      <View
+        style={{
+          flexDirection: "row",
+          flex: 1,
+          alignItems: "center",
+          justifyContent: "center",
+        }}
+      >
+        {Boolean(hasIcon) && (
+          <View style={{paddingRight: 12, paddingLeft: theme.spacing.xs as any}}>
+            <Icon color="inverted" iconName="triangle-exclamation" />
+          </View>
+        )}
+        <NativeText
+          style={{
+            color: theme.text.inverted,
+            fontWeight: "bold",
+            textAlign: "center",
+            flexShrink: 1,
+            flexWrap: "wrap",
+          }}
+        >
+          {text}
+        </NativeText>
+        {Boolean(buttonText && buttonIconName && buttonOnClick) && (
+          <View style={{paddingLeft: 16, paddingRight: theme.spacing.xs as any}}>
+            <BannerButton
+              buttonIconName={buttonIconName}
+              buttonOnClick={buttonOnClick ?? (() => {})}
+              buttonText={buttonText}
+            />
+          </View>
         )}
-      </Box>
-      {renderButton()}
-    </Box>
+        {Boolean(buttonText && !buttonIconName && buttonOnClick) && (
+          <View style={{paddingLeft: 16, paddingRight: theme.spacing.xs as any}}>
+            <BannerButton buttonOnClick={buttonOnClick ?? (() => {})} buttonText={buttonText} />
```

Comment:
I had one in there but I removed it because I made the button text required in the case that a button onclick is provided in the interface - so typescript should enforce this already

Reply:
Works for me!